### ISSUE TITLE
Timing becomes late by 12 ms after zooming into waveform

### ISSUE THREAD

**Issue by jansenmtan** [ppy/osu#18909]
### Type

Game behaviour

### Bug description

The timing is okay when first running the game, but after running the game for a while, the game judges my inputs ~11 ms late.

This issue is similar to https://github.com/ppy/osu/issues/18652, but causes late timing instead of early timing.

I suspect that the issue is related to editor timeline skipping, as mentioned in https://github.com/ppy/osu/issues/18652#issuecomment-1153800031. I usually use the editor timeline and waveform to determine the local offset of a beatmap. To do this, I zoom the timeline in as much as possible, use the Timing view's offset adjustment UI to adjust the offset to be onto the beat, and verify my adjustment by skipping around to various moments in the timeline.

Previous issues (https://github.com/ppy/osu/issues/18652#issuecomment-1153800031, https://github.com/ppy/osu/issues/18652#issuecomment-1155791046, https://github.com/ppy/osu/issues/18659#issue-1268637091) mention stuttering and lag spikes (esp. with multithreading), but stutters occur so often (maybe a few times per play) and with such little effect (a few frames at most) that I often ignore them. I will try to test with the single thread mode later and see if the issue persists.

This issue is not related to any particular beatmap.

Note: I play with effect sounds off. 

Setup/Hardware:
Windows 19044
Intel(R) Core(TM) i3-10100F CPU
Audio device: High Definition Audio Device

### Screenshots or videos

Notice that I tend to be early on this map by 1 to 4 ms, but when the bug hits, I am consistently 9 ms late.
Before bug.
![image](https://user-images.githubusercontent.com/36340429/176019474-1bee806f-ece0-4d13-8009-6d0c1a85be32.png)
I first noticed this bug during my second session of yesterday.
![image](https://user-images.githubusercontent.com/36340429/176019523-26e4acac-2c70-4d97-8d21-97dcf2d10713.png)
Today before bug.
![image](https://user-images.githubusercontent.com/36340429/176019562-2e207146-687b-4d50-9c5b-a379eb6ec3bc.png)
The bug hits mid-session.
![image](https://user-images.githubusercontent.com/36340429/176019610-a60883de-b83f-40ef-8c2b-c4866dd57fe6.png)

The bug is not limited to one map.
![image](https://user-images.githubusercontent.com/36340429/176020398-954618c2-f58c-4bb9-9b18-848118aeb2dd.png)
![image](https://user-images.githubusercontent.com/36340429/176020441-dbcabdb5-dbb2-44d1-ba47-c36aeeecb1a2.png)
![image](https://user-images.githubusercontent.com/36340429/176020535-a6f53823-007e-4dc8-8ee9-46c35c01992f.png)

### Version

2022.621.0-lazer

### Logs

[network.log](https://github.com/ppy/osu/files/8994859/network.log)
[performance.log](https://github.com/ppy/osu/files/8994860/performance.log)
[runtime.log](https://github.com/ppy/osu/files/8994861/runtime.log)
[updater.log](https://github.com/ppy/osu/files/8994862/updater.log)
[database.log](https://github.com/ppy/osu/files/8994863/database.log)
[legacy-ipc.log](https://github.com/ppy/osu/files/8994864/legacy-ipc.log)

**Comment by peppy** [comment]
Audio timing precision is only accurate to ~5am on windows, so I'm wondering if that's what you are noticing?

Generally you can solve this by adjusting your UO. But this could potentially be the same driver/whatever issue happening to a lesser extent.

To confirm, once things become delayed, does it stay that way for 100% of plays until you restart?

**Comment by jansenmtan** [comment]
The problem is not global offset or local offset.

For example, every play below the green line is on time. Not a single one of them are more than 4 ms off-time. 
However, every play above the red line (listed as 22 hrs ago, before i restarted the game) is around 11 ms late, with the exception of Kimi no Pierrot. The plays listed as 22 hrs ago _are made on the same session without any change to global or local offset_.
![image](https://user-images.githubusercontent.com/36340429/176236565-74f43b98-902a-4273-b635-442d0491520f.png)

During my sessions, my plays are never more than 5 ms off-time, so it was extremely obvious once the bug hit.

**Comment by peppy** [comment]
Going to link https://github.com/ppy/osu-stable-issues/issues/306#issuecomment-1017322498 as it may be a similar issue.

**Comment by jansenmtan** [comment]
I just found out that the bug can also revert in the same session, going back to the same timing as when the game launched.

eg, today, when the bug hit (2.5 hrs into session), I set the global offset to be 11 ms earlier to get around it. My hits for the following plays were being judged on-time after that. However, 30-40 minutes later, my hits were being judged 11-12 ms early all of a sudden.

So, adjusting the global offset is not even a good work around to this bug.

**Comment by frenzibyte** [comment]
The fact that it progressively gets out of sync (judging by your point regarding global offset not working) makes this pretty much a duplicate of https://github.com/ppy/osu/issues/12056.

**Comment by jansenmtan** [comment]
I've managed to reproduce this bug. 
https://youtu.be/bOQZbVMUpSI
As I suspected, it has to do with the editor, the editor waveform, and zooming into the editor waveform. 
The bug, when activated, causes a clear 11-12 ms timing delay (not sure what kind). The bug can be deactivated by restarting the game, after which, the 11-12 ms delay disappears.
Catching the bug is not easy. It may take around 3 or so minutes of just zooming into editor waveforms to catch the bug.
